### PR TITLE
cleaning up anchor ids, backticks, and xrefs for metering

### PR DESCRIPTION
**Pull request by lamek**
Added IDs to all modules, assemblies, and headings
Added context for xrefs
Corrected the usage of backticks.

### CODE REVIEW

File: metering/metering-about-installing-metering.adoc [hunk 1]
```diff
@@ -1,11 +1,11 @@
 [id="about-installing-metering"]
 = About installing Metering
 include::modules/common-attributes.adoc[]
-:context: about-installing-metering
+:context: metering-about-install
```
Comment (kalexand-rh) [inline metering/metering-about-installing-metering.adoc]: I'd make the filename, ID, and context statement match.

File: metering/metering-installing-metering.adoc [hunk 2]
```diff
@@ -11,6 +11,37 @@ To get started installing metering, first install the Metering Operator from Ope
 
 include::modules/metering-install-operator.adoc[leveloffset=+1]
 
-include::modules/metering-install-metering-stack.adoc[leveloffset=+1]
+// Including this content directly in the assembly because the workflow requires linking off to the config docs, and we don't current link
+// inside of modules - klamenzo 2019-09-23
+[id="metering-install-metering-stack_{context}"]
+== Installing the Metering stack 
+
+To install the metering stack, first review the xref:../metering/configuring-metering/metering-about-configuring.adoc#metering-about-configuring[configuration options] and create a `MeteringConfig` resource. You can begin the process below to generate a default `MeteringConfig`, then use the examples in the documentation to modify this default file for your specific installation.
```
Comment (kalexand-rh) [inline metering/metering-installing-metering.adoc]: s/the process below/the following process

File: metering/metering-installing-metering.adoc [hunk 3]
```diff
@@ -11,6 +11,37 @@ To get started installing metering, first install the Metering Operator from Ope
...
 
-include::modules/metering-install-metering-stack.adoc[leveloffset=+1]
+// Including this content directly in the assembly because the workflow requires linking off to the config docs, and we don't current link
+// inside of modules - klamenzo 2019-09-23
+[id="metering-install-metering-stack_{context}"]
+== Installing the Metering stack 
+
+To install the metering stack, first review the xref:../metering/configuring-metering/metering-about-configuring.adoc#metering-about-configuring[configuration options] and create a `MeteringConfig` resource. You can begin the process below to generate a default `MeteringConfig`, then use the examples in the documentation to modify this default file for your specific installation. 
+
+Review the following topics to create your `MeteringConfig`:
```
Comment (kalexand-rh) [inline metering/metering-installing-metering.adoc]: s/`MeteringConfig`:/`MeteringConfig` resource:

Comment (kalexand-rh) [inline metering/metering-installing-metering.adoc]: Rework this section to be a list of prerequisites.

File: metering/metering-installing-metering.adoc [hunk 4]
```diff
@@ -11,6 +11,37 @@ To get started installing metering, first install the Metering Operator from Ope
...
+
+* For prerequisites and general requirements, review the installation options in xref:../metering/metering-about-installing-metering.adoc#metering-about-install_metering-about-install[About installing metering].
+* For configuration options, review xref:../metering/configuring-metering/metering-about-configuring.adoc#metering-about-configuring[About configuring metering]. 
+* At a minimum, you need to xref:../metering/configuring-metering/metering-configure-persistent-storage.adoc#metering-configure-persistent-storage[configure persistent storage] and xref:../metering/configuring-metering/metering-configure-hive-metastore.adoc#metering-configure-hive-metastore[configure the Hive metastore].
```
Comment (kalexand-rh) [inline metering/metering-installing-metering.adoc]: Split this li.

File: metering/metering-using-metering.adoc [hunk 5]
```diff
@@ -8,7 +8,7 @@ toc::[]
 .Prerequisites
 
 * xref:../metering/metering-installing-metering.adoc#metering-install-operator_installing-metering[Install Metering]
-* Review the details about the available options that can be configured for a xref:../metering/metering-about-reports.adoc#metering-about-reports[Report] and how they function.
+* Review the details about the available options that can be configured for a xref:../metering/reports/metering-about-reports.adoc#metering-about-reports[Report] and how they function.
```
Comment (kalexand-rh) [inline metering/metering-using-metering.adoc]: Why is this a separate assembly?

File: modules/metering-debugging.adoc [hunk 6]
```diff
@@ -3,8 +3,8 @@
 // * metering/metering-troubleshooting-debugging.adoc
 
 [id="metering-debugging_{context}"]
-
 = Debugging metering
+
 Debugging metering is much easier when you interact directly with the various components. The sections below detail how you can connect and query Presto and Hive, as well as view the dashboards of the Presto and HDFS components.
```
Comment (kalexand-rh) [inline modules/metering-debugging.adoc]: no comma

File: modules/metering-exposing-the-reporting-api.adoc [hunk 7]
```diff
@@ -25,6 +25,7 @@ METERING_ROUTE_HOSTNAME=$(oc -n openshift-metering get routes metering -o json |
 
 Next, set up authentication using either a service account token or basic authentication with a username/password.
 
+[id="metering-authenticate-using-service-account_{context}"]
```
Comment (kalexand-rh) [inline modules/metering-exposing-the-reporting-api.adoc]: What is this module supposed to be? It's not a well-formed procedure module - that would have a single heading and a clear procedure section. Only procedure modules are allowed to have gerunds in the module title.

File: metering/metering-about-installing-metering.adoc (hunk 1 above)
Comment (vikram-redhat) [reply metering/metering-about-installing-metering.adoc]: True - it is a requirement for id and context and a suggestion for filename: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata

File: modules/metering-exposing-the-reporting-api.adoc [hunk 8]
```diff
@@ -36,10 +37,11 @@ curl -H "Authorization: Bearer $TOKEN" -k "https://$METERING_ROUTE_HOSTNAME/api/
 
 Be sure to replace the `name=[Report Name]` and `format=[Format]` parameters in the URL above. The `format` parameter can be json, csv, or tabular.
 
+[id="metering-authenticate-using-username-password_{context}"]
 === Authenticate using a username and password
 We are able to do basic authentication using a username and password combination, which is specified in the contents of a htpasswd file. By default, we create a secret containing an empty htpasswd data. You can, however, configure the `reporting-operator.spec.authProxy.htpasswd.data` and `reporting-operator.spec.authProxy.htpasswd.createSecret` keys to use this method.
 
-Once you have specified the above in your `MeteringConfig`, you can run the following command:
+Once you have specified the above in your MeteringConfig, you can run the following command:
```
Comment (kalexand-rh) [inline modules/metering-exposing-the-reporting-api.adoc]: s/Once you have specified the above in your MeteringConfig/After you specify the required values in the MeteringConfig definition

File: modules/metering-exposing-the-reporting-api.adoc [hunk 9]
```diff
@@ -48,8 +50,9 @@ curl -u testuser:password123 -k "https://$METERING_ROUTE_HOSTNAME/api/v1/reports
 
 Be sure to replace `testuser:password123` with a valid username and password combination.
 
+[id="metering-manually-configure-authentication_{context}"]
 == Manually Configuring Authentication
-In order to manually configure, or disable OAuth in the `reporting-operator`, you must set `spec.tls.enabled: false` in your `MeteringConfig`.
+In order to manually configure, or disable OAuth in the `reporting-operator`, you must set `spec.tls.enabled: false` in your MeteringConfig.
```
Comment (kalexand-rh) [inline modules/metering-exposing-the-reporting-api.adoc]: ~In order~
no comma
MeteringConfig resource

File: modules/metering-exposing-the-reporting-api.adoc [hunk 10]
```diff
@@ -71,6 +74,7 @@ You can generate a 32-character random string using the command below.
 $ openssl rand -base64 32 | head -c32; echo.
 ----
 
+[id="metering-token-authentication_{context}"]
 === Token Authentication
```
Comment (kalexand-rh) [inline modules/metering-exposing-the-reporting-api.adoc]: sentence case

File: modules/metering-exposing-the-reporting-api.adoc [hunk 11]
```diff
@@ -71,6 +74,7 @@ You can generate a 32-character random string using the command below.
...
 When the following options are set to `true`, authentication using a bearer token is enabled for the reporting REST API. Bearer tokens may come from serviceAccounts or users.
```
Comment (kalexand-rh) [inline modules/metering-exposing-the-reporting-api.adoc]: s/may/can

File: modules/metering-exposing-the-reporting-api.adoc [hunk 12]
```diff
@@ -85,7 +89,7 @@ When authentication is enabled, the Bearer token used to query the reporting API
 * metering-admin
 * metering-viewer
 
-The `metering-operator` is capable of creating `RoleBindings` for you, granting these permissions by specifying a list of subjects in the `spec.permissions` section. For an example, see the `advanced-auth.yaml` example configuration below.
+The `metering-operator` is capable of creating RoleBindings for you, granting these permissions by specifying a list of subjects in the `spec.permissions` section. For an example, see the `advanced-auth.yaml` example configuration below.
```
Comment (kalexand-rh) [inline modules/metering-exposing-the-reporting-api.adoc]: Metering Operator or `metering-operator` resource - you need to qualify objects with their type.
never "below"
> ! below adj
Do not use to indicate a relative location in a document, as in “the information below.” Use “the information
later in this section.”
Alternatively, you can use “following” to refer to a table, to art, or to a list. Example: “The following table
shows ...”

File: modules/metering-install-verify.adoc [hunk 13]
```diff
@@ -61,4 +61,4 @@ pod-usage-cpu-cores                          2019-08-05T16:52:00Z   2019-08-05T1
 pod-usage-memory-bytes                       2019-08-05T16:52:00Z   2019-08-05T18:08:00Z   2019-08-05T16:52:00Z   2019-08-05T18:08:00Z   2019-08-05T18:54:20Z   9m49s
 ----
 
-Once all Pods are ready and you have verified that data is being imported, you can begin using Metering to collect data and report on your cluster.
+Once all Pods are ready and you have verified that data is being imported, you can begin using metering to collect data and report on your cluster.
```
Comment (kalexand-rh) [inline modules/metering-install-verify.adoc]: s/Once/After
> ! once adv
Use only to mean “one time.” Do not use as a conjunction to mean “after” or “when.”

File: modules/metering-prometheus-connection.adoc [hunk 14]
```diff
@@ -7,7 +7,7 @@
 
 When you install metering on {product-title}, Prometheus is available at https://prometheus-k8s.openshift-monitoring.svc:9091/.
 
-To secure the connection to Prometheus, the default metering installation uses the OpenShift certificate authority. If your Prometheus instance uses a different CA, the CA can be injected through a `ConfigMap`. See the example below.
+To secure the connection to Prometheus, the default metering installation uses the OpenShift certificate authority. If your Prometheus instance uses a different CA, the CA can be injected through a ConfigMap. See the example below.
```
Comment (kalexand-rh) [inline modules/metering-prometheus-connection.adoc]: Never "OpenShift" by itself. s/OpenShift/{product-title}
You can use OpenShift as part of a specific component name if you must.

Comment (kalexand-rh) [reply modules/metering-prometheus-connection.adoc]: (never "below")

File: modules/metering-reports.adoc [hunk 15]
```diff
@@ -4,15 +4,16 @@
 [id="metering-reports_{context}"]
 = Reports
 
-The `Report` custom resource is used to manage the execution and status of reports.
+The Report custom resource is used to manage the execution and status of reports.
 Metering produces reports derived from usage data sources, which can be used in further analysis and filtering.
 
-A single `Report` resource represents a job manages a database table and updates it with new information according to a schedule, and exposes the data in that table via the reporting-operator HTTP API.
-Reports with a `spec.schedule` field set are always running, and will track what time periods it has collected data for, ensuring that if Metering is shutdown or unavailable for an extended period of time, it will backfill the data starting where it left off.
+A single Report resource represents a job manages a database table and updates it with new information according to a schedule, and exposes the data in that table via the reporting-operator HTTP API.
```
Comment (kalexand-rh) [inline modules/metering-reports.adoc]: no comma

File: modules/metering-reports.adoc [hunk 16]
```diff
@@ -4,15 +4,16 @@
...
+Reports with a `spec.schedule` field set are always running, and will track what time periods it has collected data for, ensuring that if metering is shutdown or unavailable for an extended period of time, it will backfill the data starting where it left off.
```
Comment (kalexand-rh) [inline modules/metering-reports.adoc]: no comma after running
Don't use future tense unless it's critical to the clarity of the passage.

File: modules/metering-reports.adoc [hunk 17]
```diff
@@ -4,15 +4,16 @@
...
 If the schedule is unset, then the Report will run once for the time specified by the `reportingStart` and `reportingEnd`.
-By default, reports will wait for `ReportDataSources` to have the progressed in their import process to cover the report period being processed.
+By default, reports will wait for ReportDataSources to have the progressed in their import process to cover the report period being processed.
```
Comment (kalexand-rh) [inline modules/metering-reports.adoc]: Please clarify this sentence.

File: modules/metering-reports.adoc [hunk 18]
```diff
@@ -48,14 +50,15 @@ spec:
   reportingEnd: "2019-07-31T00:00:00Z"
 ----
 
+[id="metering-query_{context}"]
```
Comment (kalexand-rh) [inline modules/metering-reports.adoc]: Consider adding some transitional language here - I'm not sure how we got from sample reports to "query." Is this the start of some command or parameter examples? Do you need another H2 and to make this heading and the following headings H3s? (If this is reference content, consider moving it to its own module.)

File: modules/metering-reports.adoc [hunk 19]
```diff
@@ -107,19 +110,19 @@ pod-memory-usage                             23m
 pod-memory-usage-raw                         23m
 ----
 
-`ReportQueries` with the `-raw` suffix are used by other `ReportQueries` to build more complex queries, and should not be used directly for reports.
+ReportQueries with the `-raw` suffix are used by other ReportQueries to build more complex queries, and should not be used directly for reports.
 
 `namespace-` prefixed queries aggregate Pod CPU/memory requests by namespace, providing a list of namespaces and their overall usage based on resource requests.
 
-`pod-` prefixed queries are similar to 'namespace-' prefixed, but aggregate information by Pod, rather than namespace. These queries include the Pod's namespace and node.
+`pod-` prefixed queries are similar to `namespace-` prefixed queries, but aggregate information by Pod rather than namespace. These queries include the Pod's namespace and node.
```
Comment (kalexand-rh) [inline modules/metering-reports.adoc]: no comma

File: modules/metering-store-data-in-shared-volumes.adoc [hunk 20]
```diff
@@ -10,9 +10,9 @@
 NFS is not recommended to use with metering. 
 ====
 
-Metering has no storage by default, but it can use any `ReadWriteMany PersistentVolume` or any `StorageClass` that provisions a `ReadWriteMany PersistentVolume`.
+Metering has no storage by default, but it can use any ReadWriteMany PersistentVolume or any StorageClass that provisions a ReadWriteMany PersistentVolume.
 
-To use a `ReadWriteMany PersistentVolume` for storage, modify the `shared-storage.yaml` file below.
+To use a ReadWriteMany PersistentVolume for storage, modify the `shared-storage.yaml` file below.
```
Comment (kalexand-rh) [inline modules/metering-store-data-in-shared-volumes.adoc]: Add ".Procedure"

Comment (kalexand-rh) [reply modules/metering-store-data-in-shared-volumes.adoc]: The step in a single-step procedure needs to be presented as a unordered list item.

File: modules/metering-store-data-in-shared-volumes.adoc [hunk 21]
```diff
@@ -36,6 +36,6 @@ spec:
 
 There are two options to modify your configuration file:
```
Comment (kalexand-rh) [inline modules/metering-store-data-in-shared-volumes.adoc]: I'd remove this line.

File: modules/metering-store-data-in-shared-volumes.adoc [hunk 22]
```diff
@@ -36,6 +36,6 @@ spec:
...
 
-<1> Set `storage.hive.sharedPVC.claimName` to the name of an existing `ReadWriteMany` persistent volume claim (PVC). This is necessary if you do not have dynamic volume provisioning, or want to have more control over how the `PersistentVolume` is created.
+<1> Set `storage.hive.sharedPVC.claimName` to the name of an existing ReadWriteMany PersistentVolumeClaim (PVC). This is necessary if you do not have dynamic volume provisioning, or want to have more control over how the PersistentVolume is created.
```
Comment (kalexand-rh) [inline modules/metering-store-data-in-shared-volumes.adoc]: I'd make it clear in these annotations that you need to pick one one.
(no comma)

File: modules/metering-troubleshooting.adoc [hunk 23]
```diff
@@ -5,7 +5,7 @@
 [id="metering-troubleshooting_{context}"]
 = Troubleshooting metering
 
-A common issue with metering is Pods failing to start. Pods may fail to start due to lack of resources, or if they have a dependency on a resource that does not exist, such as a `StorageClass` or `Secret`.
+A common issue with metering is Pods failing to start. Pods may fail to start due to lack of resources, or if they have a dependency on a resource that does not exist, such as a StorageClass or Secret.
```
Comment (kalexand-rh) [inline modules/metering-troubleshooting.adoc]: s/may/might
no comma

File: modules/metering-writing-reports.adoc [hunk 24]
```diff
@@ -1,12 +1,12 @@
...
 // * metering/metering-using-metering.adoc
-[id="metering-writing-reports"]
+[id="metering-writing-reports_{context}"]
 = Writing Reports
 
-Writing a `Report` is the way to process and analyze data using Metering.
+Writing a Report is the way to process and analyze data using Metering.
 
-To write a report, you must define a `Report` resource in a yaml file, and specify the required parameters, and create it in the `openshift-metering` namespace using `oc`.
+To write a Report, you must define a Report resource in a YAML file, and specify the required parameters, and create it in the `openshift-metering` namespace using `oc`.
```
Comment (kalexand-rh) [inline modules/metering-writing-reports.adoc]: ~and~ specify
s/using/by using

File: modules/metering-writing-reports.adoc [hunk 25]
```diff
@@ -20,7 +20,7 @@ To write a report, you must define a `Report` resource in a yaml file, and speci
 $ oc project openshift-metering
 ----
 
-. Create a report resources as a YAML file:
+. Create a Report resources as a YAML file:
 +
 .. Create a YAML file with the following:
```
Comment (kalexand-rh) [inline modules/metering-writing-reports.adoc]: The following what? You need a noun here.

File: modules/metering-writing-reports.adoc [hunk 26]
```diff
@@ -36,11 +36,11 @@ spec:
   query: namespace-cpu-request <1>
   runImmediately: true <3>
 ----
-<1> The `query` specifies `ReportQuery` used to generate the report. Change this based on what you want to report on. For a list of options, run `oc get reportqueries | grep -v raw`.
-<2> The `metadata.name` of the report should be something descriptive about what the report does. A good name is the query, and the schedule or period you used.
+<1> The `query` specifies ReportQuery used to generate the Report. Change this based on what you want to Report on. For a list of options, run `oc get reportqueries | grep -v raw`.
```
Comment (kalexand-rh) [inline modules/metering-writing-reports.adoc]: s/Report/report - you mean the verb not the API object.

File: modules/metering-writing-reports.adoc [hunk 27]
```diff
@@ -36,11 +36,11 @@ spec:
...
+<2> The `metadata.name` of the Report should be something descriptive about what the Report does. A good name is the query, and the schedule or period you used.
```
Comment (kalexand-rh) [inline modules/metering-writing-reports.adoc]: Rewrite to not use "should."
> ! should v
Always prefer a more direct verb or way of saying something. Omit “should,” or use “must” or “might,” as
appropriate.

File: metering/metering-installing-metering.adoc (hunk 3 above)
Comment (lamek) [reply metering/metering-installing-metering.adoc]: Good call.

File: metering/metering-using-metering.adoc (hunk 5 above)
Comment (lamek) [reply metering/metering-using-metering.adoc]: The "Using metering" section walks you through the basics of creating and viewing a report. These are the two main ways a user can interact with metering. The "Reports" section is more referential, containing all the possible things you can do with reports. I still have a few topics to add here in the Reports section.

File: modules/metering-debugging.adoc (hunk 6 above)
Comment (lamek) [reply modules/metering-debugging.adoc]: I'm always a fan of fewer commas. :clinking_glasses:

File: modules/metering-exposing-the-reporting-api.adoc (hunk 7 above)
Comment (lamek) [reply modules/metering-exposing-the-reporting-api.adoc]: Good point. Upstream you had some options around how to expose the reporting API, but the need to cover those goes away on OS b/c the route is automatically exposed. I'll rework this, I think to be "Accessing the reporting API", and then I'll update the ensuing content to be more procedural.

File: modules/metering-reports.adoc (hunk 15 above)
Comment (lamek) [reply modules/metering-reports.adoc]: :raised_hands: I split this into two sentences.

File: modules/metering-reports.adoc (hunk 16 above)
Comment (lamek) [reply modules/metering-reports.adoc]: I updated this, also split the sentence.

File: modules/metering-reports.adoc (hunk 17 above)
Comment (lamek) [reply modules/metering-reports.adoc]: Reports wait for the necessary data to import before running. I updated these two sentences. Good call.

**Pull request merged**